AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/MutationQueue/SyncMutationToCloudOperationTests.swift

```diff
@@ -19,12 +19,14 @@ class SyncMutationToCloudOperationTests: XCTestCase {
     let defaultAsyncWaitTimeout = 2.0
     let secondsInADay = 60 * 60 * 24
     var mockAPIPlugin: MockAPICategoryPlugin!
-    var networkReachability: NetworkReachabilityNotifier!
+
+    var reachabilityPublisher: PassthroughSubject<ReachabilityUpdate, DataStoreError>!
```

Comment:
these need to be changed over to APIErrors

Reply:
Looked into reachability errors -- the errors seem to occur during instantiation and attempting to start the reachability subsystem so... In the following Pr:
https://github.com/aws-amplify/amplify-ios/pull/269

I made two changes:
1.  I have coupled the instantiation & starting of notifier into a single success (or failure) criteria -- if they both succeed, we return an instance -- if one of them fails, then we return nil.
2.  Change the PassthroughSubject from emitting errors to Never.